lib/sidekiq_smart_cache/redis.rb

```diff
@@ -53,12 +53,34 @@ def method_missing(name, *args)
                 nil # quietly return nil in this case
               end
             else
+              # Dealing with the issue of the last argument being keyword arguments
```

Comment:
I think we can use argument forwarding with leading arguments, which was introduced in Ruby 3: https://rubyreferences.github.io/rubychanges/3.0.html#arguments-forwarding--supports-leading-arguments
It does seem to have been backported to our version or Ruby -- I've been playing with argument forwarding with leading arguments locally on ruby 2.7.5 and it's been working as I expect.
An extremely rough cut (not at all tested), but I think you could do something like this:
```
def method_missing(name, ...)
  @pool.with do |r|
    if COMMANDS.include? name
      retryable = true
      begin
      	# The commented out logging below is not 100% feasible with argument forwarding
        # log_msg("#{name} #{args}")
        if r.respond_to?(name)
          # old redis gem implements methods including `brpop` and `flusdb`
          r.send(name, ...)
        elsif BLOCKING_COMMANDS.include? name
          # support redis-client semantics
          make_blocking_call(r, name, ...)
        else
          r.call(name.to_s.upcase, ...)
        end
        # stolen from sidekiq - Thanks Mike!
        # Quietly consume this one and return nil
      rescue ERROR_TO_CATCH => ex
        # 2550 Failover can cause the server to become a replica, need
        # to disconnect and reopen the socket to get back to the primary.
        if retryable && ex.message =~ /READONLY/
          r.disconnect!
          retryable = false
          retry
        end
        raise
      end
    else
      super
    end
  end

  def make_blocking_call(r, name, *args)
  	r.blocking_call(args[1], name.to_s.upcase, args[0], 0)
  rescue ::RedisClient::TimeoutError
    nil # quietly return nil in this case
  end
 end
 ```
 The work around to get at the positional args for constructing the blocking_call is a little weird, but I think it's worth it to take advantage of argument forwarding as a very simple way to deal with the keyword args.

Reply:
ooh! I hadn't processed that `...` was available in 2.7.5, this is much better.

That implementation only supports a single queue, and `brpop` can (in theory) pull from multiple.  I did the more generic implementation, for fewer surprises down the line.

Reply:
Sweet! 🙌